### PR TITLE
Add voting portal header + stubs of all the pages

### PR DESCRIPTION
**Pull request by Will-Howard**
See [designs](https://www.figma.com/file/OWruitRTyWpmLJjaT4RedA/Voting-portal?node-id=21%3A12808&mode=dev) for the voting portal here

This PR adds the header (mostly functional, just needs some user state to determine which links to enable), and admin-gated stubs of all the pages (with none of the content, or backend functionality). The purpose of this is so people can work on the different pages independently with less duplication

Screenshot of the header:
Screenshot 2023-11-22 at 14 29 06

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206011464269056) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/ea-forum/voting-portal/VotingPortalHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+  NavigationMenuButton: FC,
+  RightHeaderItems: FC,
+  HeaderNavigationDrawer: FC,
+  HeaderNotificationsMenu: FC,
+  classes: ClassesType,
+}) => {
+  const { Typography, HeadTags } = Components;
+  const { pathname } = useLocation();
+
+  return (
```
Comment (oetherington) [inline packages/lesswrong/components/ea-forum/voting-portal/VotingPortalHeader.tsx:203]: I'm not sure this is worth making a separate component vs just integrating direcctly into `GivingSeasonHeader` with a `pathname` check (I don't think we should worry about it getting messy for such a short-lived component). My main worry is that there are known bugs with this already which will now need to be fixed in two places (see #8293 for instance which would now need to be copied into this file).

**Review by oetherington — APPROVED** [review]
Looks good!

Do we specifically need to deploy this to prod for testing? If not then it'd probably be more straight forward to remove all the admin gates now and just merge to a feature branch? (And maybe deploy to beta?)

**Comment by oetherington** [comment]
Also just noticed this:
Screenshot 2023-11-23 at 13 10 51
 
After you click that button it stays in the "hovered" state with less opacity.

Also, I don't like that there's no button to go back to the homepage without opening the side menu - maybe the lightbulb should link back there?

**Comment by Will-Howard** [comment]
>Do we specifically need to deploy this to prod for testing?

No, although I have a slight preference for doing this to reduce the amount of conflict resolving as more stuff is merged in. For instance there will be a few migrations required to store the votes and merging directly to master will save fixing ordering issues

File: packages/lesswrong/components/common/Header.tsx [hunk 2]
```diff
@@ -73,7 +74,7 @@ export const styles = (theme: ThemeType): JssStyles => ({
   titleLink: {
     color: theme.palette.header.text,
     fontSize: 19,
-    '&:hover, &:focus, &:active': {
+    '&:hover, &:active': {
```
Comment (Will-Howard) [inline packages/lesswrong/components/common/Header.tsx:77]: Fix for the hover issue, this was actually a long term bug pre giving season

**Comment by Will-Howard** [comment]
> Also, I don't like that there's no button to go back to the homepage without opening the side menu - maybe the lightbulb should link back there?

I actually decided there wasn't really any need to have the ability to navigate back to `/voting-portal`, so I changed it to go to the frontpage as usual. This is slightly confusing, and it might be good to reword it like "EA Forum | Voting portal" or similar, but I'll leave that up to @agnesstenlund for when more of this is done

**Pull request merged**